ProductManagement/Entity/Product.cs

```diff
@@ -0,0 +1,56 @@
+﻿using MongoDB.Bson.Serialization.Attributes;
+using MongoDB.Bson;
+
+namespace SimpleInventoryManagementSystem.ProductManagement.Entity
+{
+    public class Product
+    {
+        [BsonId]
+        [BsonRepresentation(BsonType.ObjectId)]
+        public ObjectId MongoId { get; set; }
+
+        [BsonElement("Id")]
+        public int Id { get; set; }
```

Comment:
Why not use the same identifier for both MongoDb and SQL? I realize that auto incremeding fields in MongoDb can lead to some performance issues when using `findAndModify()`, but how about just using an ID that is supported by both like a UUID.

Reply:
Agree! Done it.